src/udev/udevd.c

```diff
@@ -1490,10 +1498,12 @@ static int on_post(sd_event_source *s, void *userdata) {
         if (manager->exit)
                 return sd_event_exit(manager->event, 0);
 
-        if (manager->cgroup)
+        if (manager->cgroup && !manager->ctrl_is_connected)
```

Comment:
won't this risk leaving something behind if it happens to be running at the same time as udevadm?

Reply:
Potentially yes. But `on_post()` is frequently called. So hopefully they are killed later soon.